### PR TITLE
Fix true/false not rendering properly in yarn.lock files

### PR DESCRIPTION
**Pull request by Validark**
I am not sure at this moment how to make a test for this but in an empty folder the following commands repro the issue.

```
bun init
bun add true -y
```

At the time of writing this produces a yarn.lock file which looks like so:
```
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1
# bun ./bun.lockb --hash: 11FD1BB989596431-20acd09908802a90-46C46CAD6336683E-4a23890a7915a917


bun-types@^0.5.0:
  version "0.5.8"
  resolved "https://registry.npmjs.org/bun-types/-/bun-types-0.5.8.tgz"
  integrity sha512-VHwD0MAHo3wraYAeqTWH2NDmXOdGfC3wWWOnZvK93ytI6yq/LkgsCjDudWNmN7MlfPvJb2zoLMnkzhjxNwLsLw==

true@:
  version "0.0.4"
  resolved "https://registry.npmjs.org/true/-/true-0.0.4.tgz"
  integrity sha512-e7hkFCda5y51OTE4eQHVPFbuljGnOpzrVcigGM5Pd9+Bw1bEsHCq3mVIm/z/O5Mh1bXtVPNxUV8oCl/dMunlRg==
```

This PR fixes two bugs with the `true` emit. First of all, any package beginning with "true" or "false" would be wrapped in quotation marks by yarn ([source](https://github.com/yarnpkg/yarn/blob/158d96dce95313d9a00218302631cd263877d164/src/lockfile/stringify.js#L11)). Second of all, the version is not written between its `@` and `:`. I was not sure if this should have been fixed earlier in the pipeline but it appears to only happen to packages you installed with `bun add` or `bun install`. This makes it just drop the resolved version with a '^' in front of it. This seems correct in the most common case but again, I am not sure if it would more elegantly be fixed in a different way.

### CODE REVIEW

File: src/install/lockfile.zig [hunk 1]
```diff
@@ -1225,7 +1233,11 @@ pub const Printer = struct {
 
                         try writer.writeAll(name);
                         try writer.writeByte('@');
-                        try writer.writeAll(version_name);
+                        if (version_name.len == 0) {
+                            try std.fmt.format(writer, "^{any}", .{version_formatter});
```
Comment (Jarred-Sumner) [inline src/install/lockfile.zig:1237]: They might change `any` to mean "use the builtin std.fmt.format" in https://github.com/ziglang/zig/pull/15093

so maybe this should be `{}` 

```suggestion
                            try std.fmt.format(writer, "^{}", .{version_formatter});
```

Comment (Jarred-Sumner) [reply src/install/lockfile.zig:1237]: but we can make that change in batch later to everywhere in the codebase I guess

**Comment by Jarred-Sumner** [comment]
Thank you. In the future please include tests

**Comment by Electroid** [comment]
@robobun thoughts?

**Comment by Electroid** [comment]
@robobun !

**Comment by Electroid** [comment]
@robobun !!

**Comment by Electroid** [comment]
@robobun 
```js
console.log("Hello!");
```

**Pull request merged**